src/System/System.php

```diff
@@ -128,4 +157,360 @@ static public function isArch(string $arch): bool
+            $data[$cpuNumber]['idle'] = $cpu[4];
+            $data[$cpuNumber]['iowait'] = $cpu[5];
+            $data[$cpuNumber]['irq'] = $cpu[6];
+            $data[$cpuNumber]['softirq'] = $cpu[7];
+
+            // These might not exist on older kernels.
+            $data[$cpuNumber]['steal'] = isset($cpu[8]) ? $cpu[8] : 0;
+            $data[$cpuNumber]['guest'] = isset($cpu[9]) ? $cpu[9] : 0;
+        }
+
+        if (!$totalCPUExists) {
+            // Combine all values
+            $data['total'] = [
+                'user' => 0,
+                'nice' => 0,
+                'system' => 0,
+                'idle' => 0,
+                'iowait' => 0,
+                'irq' => 0,
+                'softirq' => 0,
+                'steal' => 0,
+                'guest' => 0
+            ];
+
+            foreach ($data as $cpu) {
+                $data['total']['user'] += $cpu['user'];
+                $data['total']['nice'] += $cpu['nice'];
+                $data['total']['system'] += $cpu['system'];
+                $data['total']['idle'] += $cpu['idle'];
+                $data['total']['iowait'] += $cpu['iowait'];
+                $data['total']['irq'] += $cpu['irq'];
+                $data['total']['softirq'] += $cpu['softirq'];
+                $data['total']['steal'] += $cpu['steal'];
+                $data['total']['guest'] += $cpu['guest'];
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Gets the current usage of a core as a percentage. Passing 0 will return the usage of all cores combined.
+     * 
+     * @param int $core
+     * 
+     * @return int
+     * 
+     * @throws Exception
+     */
+    static public function getCPUUtilisation(int $id = 0): int
+    {
+        switch (self::getOS()) {
+            case 'Linux':
+                $cpuNow = self::getProcStatData();
+                $i = 0;
+
+                $data = [];
+
+                foreach ($cpuNow as $cpu) {
+                    // Check if this is the total CPU
+                    $cpuTotal = $cpu['user'] + $cpu['nice'] + $cpu['system'] + $cpu['idle'] + $cpu['iowait'] + $cpu['irq'] + $cpu['softirq'] + $cpu['steal'];
+
+                    $cpuIdle = $cpu['idle'];
+
+                    $idleDelta = $cpuIdle - (isset($lastData[$i]) ? $lastData[$i]['idle'] : 0);
+
+                    $totalDelta = $cpuTotal - (isset($lastData[$i]) ? $lastData[$i]['total'] : 0);
+
+                    $lastData[$i]['total'] = $cpuTotal;
+                    $lastData[$i]['idle'] = $cpuIdle;
+
+                    $result = (1.0 - ($idleDelta / $totalDelta)) * 100;
+
+                    $data[$i] = $result;
+
+                    $i++;
+                }
+
+                if ($id === 0) {
+                    return intval(array_sum($data));
+                } else {
+                    return $data[$id];
+                }
+            default:
+                throw new Exception(self::getOS() . " not supported.");
```

Comment:
Would be nice to add a table covering all the different method that are not supported in all operating systems.